### PR TITLE
Return no updates when performing an NX set to an existing array element

### PR DESCRIPTION
**Pull request by EricRabil**
This PR addresses https://github.com/RedisJSON/RedisJSON/issues/592, although I'd like some feedback from other contributors on how I fixed it as I'm not sure if this is intentional behavior.

This does not modify the mutations made to the document, but instead stops returning an out-of-range error when doing an NX set to an array element that is, in fact, in range.

### CODE REVIEW

**Comment by oshadmi** [comment]
Related [comment](https://github.com/RedisJSON/RedisJSON/pull/336#discussion_r661200127)

**Comment by oshadmi** [comment]
@EricRabil Thank you for your fix. It looks good. We will be happy to merge it once tests are added.
Please let us know if you are available to add tests (if not, we will add)

**Comment by EricRabil** [comment]
Awesome. I'll add tests in the next day

**Comment by EricRabil** [comment]
@oshadmi tests have been added, let me know if you'd like any changes to be made to them.

File: tests/pytest/test.py [hunk 1]
```diff
@@ -925,6 +925,11 @@ def testMultiPathResults(env):
     # make sure legacy json path returns single result
     env.expect("JSON.GET", "k", '.*[0,2]').equal('1')
 
+def testIssue_597(env):
+    env.expect("JSON.SET", "test", ".", "[0]").ok()
+    env.expect("JSON.SET", "test", ".[0]", "[0]", "NX").noError()
+    env.expect("JSON.SET", "test", ".[1]", "[0]", "NX").raiseError()
```
Comment (oshadmi) [inline tests/pytest/test.py:931]:
```suggestion
    env.assertEqual(env.execute_command("JSON.SET", "test", ".[0]", "[0]", "NX"), None)
    env.expect("JSON.SET", "test", ".[1]", "[0]", "NX").raiseError()
    # make sure value was not changed
    env.expect("JSON.GET", "test", ".").equal('[0]')
```

**Review by oshadmi — COMMENTED** [review]
Minor addition.
Thanks!

**Comment by EricRabil** [comment]
@oshadmi do you have any additional changes for me to make?

**Review by oshadmi — APPROVED** [review]
👍🏼

**Pull request merged**